### PR TITLE
add scalafmt sbt plugin

### PR DESCRIPTION
**Pull request by imkarrer**
does what it says on the tin.  I couldn't figure out how to get IJ to work with scala3 very well and need to use my command line to format.

github actions can use it to toss out unformatted prs in the future too.

### CODE REVIEW

**Comment by hejfelix** [comment]
Cool, can you add the check to `.github/workflows/scala.yml` ?

**Comment by imkarrer** [comment]
I think `aa4cd32` may have failed because I am consuming java 17 locally instead of java 11 like the build does.  git says my local is the same as the remote copy, and the formatting tasks work. (getting new workstation setup this year and still configuring things)

File: modules/slf4j/src/main/scala/org/legogroup/woof/slf4j/WoofLogger.scala [hunk 1]
```diff
@@ -15,7 +15,8 @@ class WoofLogger(name: String) extends Logger:
   private def getLogInfo() =
     val stacktraceElements = Thread.currentThread().getStackTrace()
     val callingMethodIndex =
-      stacktraceElements.size - stacktraceElements.reverse.indexWhere(s => s.getClassName == this.getClass.getName,
+      stacktraceElements.size - stacktraceElements.reverse.indexWhere(s =>
+        s.getClassName == this.getClass.getName,
```
Comment (imkarrer) [inline modules/slf4j/src/main/scala/org/legogroup/woof/slf4j/WoofLogger.scala:19]: The sbt cli command `sbt scalafmt` prefers this on my local workstation, but both IJ and the build do not 🧑‍🔬 Ill figure it out.

Comment (imkarrer) [reply modules/slf4j/src/main/scala/org/legogroup/woof/slf4j/WoofLogger.scala]: I swapped it back and it still doesn't like it....

Comment (imkarrer) [reply modules/slf4j/src/main/scala/org/legogroup/woof/slf4j/WoofLogger.scala:19]: it doesn't seem to matter what state the file is in, ci doesn't like it...

It appears that it is running against the merge commit, and not the PR.  So the post merge format is not okay...

3307dd9a25d63d9551682a16d7d53b3335303a72 is the current failed commit id it tried.

Comment (imkarrer) [reply modules/slf4j/src/main/scala/org/legogroup/woof/slf4j/WoofLogger.scala:19]: wtf, you seeing this? I think I have found an un deterministic bug in scalafmt.  I just clean and run the formatter, and it keeps going back and forth, throwing out the change that was just formatted.

- `sbt clean`
- `sbt scalafmt` <- formats to two lines
- `sbt scalafmt` <- does nothing
- `sbt clean`<- formats
- `sbt scalafmt` <- formats it back the other way onto a single line

```
imkarrer@IsaacsWorkstation woof % sbt scalafmt
[info] welcome to sbt 1.6.1 (Homebrew Java 11.0.12)
[info] loading settings for project woof-build from plugins.sbt ...
[info] loading project definition from /Users/imkarrer/github/woof/project
[info] loading settings for project root from build.sbt ...
[info] set current project to root (in build file:/Users/imkarrer/github/woof/)
[info] scalafmt: Formatting 1 Scala sources (/Users/imkarrer/github/woof/modules/http4s/.jvm)...
[info] scalafmt: Formatting 14 Scala sources (/Users/imkarrer/github/woof/modules/core/jvm)...
[info] scalafmt: Formatting 2 Scala sources (/Users/imkarrer/github/woof/modules/slf4j)...
[info] scalafmt: Formatting 1 Scala sources (/Users/imkarrer/github/woof/modules/http4s/.js)...
[info] scalafmt: Formatting 14 Scala sources (/Users/imkarrer/github/woof/modules/core/js)...
[info] scalafmt: Reformatted 1 Scala sources
[success] Total time: 2 s, completed Jan 12, 2022, 12:52:45 PM
imkarrer@IsaacsWorkstation woof % git status
On branch add-scalafmt-sbt-plugin
Your branch is up to date with 'origin/add-scalafmt-sbt-plugin'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   modules/slf4j/src/main/scala/org/legogroup/woof/slf4j/WoofLogger.scala

no changes added to commit (use "git add" and/or "git commit -a")
imkarrer@IsaacsWorkstation woof % sbt scalafmt
[info] welcome to sbt 1.6.1 (Homebrew Java 11.0.12)
[info] loading settings for project woof-build from plugins.sbt ...
[info] loading project definition from /Users/imkarrer/github/woof/project
[info] loading settings for project root from build.sbt ...
[info] set current project to root (in build file:/Users/imkarrer/github/woof/)
[success] Total time: 1 s, completed Jan 12, 2022, 12:53:31 PM
imkarrer@IsaacsWorkstation woof % sbt clean   
[info] welcome to sbt 1.6.1 (Homebrew Java 11.0.12)
[info] loading settings for project woof-build from plugins.sbt ...
[info] loading project definition from /Users/imkarrer/github/woof/project
[info] loading settings for project root from build.sbt ...
[info] set current project to root (in build file:/Users/imkarrer/github/woof/)
[success] Total time: 0 s, completed Jan 12, 2022, 12:53:38 PM
imkarrer@IsaacsWorkstation woof % sbt scalafmt
[info] welcome to sbt 1.6.1 (Homebrew Java 11.0.12)
[info] loading settings for project woof-build from plugins.sbt ...
[info] loading project definition from /Users/imkarrer/github/woof/project
[info] loading settings for project root from build.sbt ...
[info] set current project to root (in build file:/Users/imkarrer/github/woof/)
[info] scalafmt: Formatting 14 Scala sources (/Users/imkarrer/github/woof/modules/core/js)...
[info] scalafmt: Formatting 14 Scala sources (/Users/imkarrer/github/woof/modules/core/jvm)...
[info] scalafmt: Formatting 1 Scala sources (/Users/imkarrer/github/woof/modules/http4s/.jvm)...
[info] scalafmt: Formatting 2 Scala sources (/Users/imkarrer/github/woof/modules/slf4j)...
[info] scalafmt: Formatting 1 Scala sources (/Users/imkarrer/github/woof/modules/http4s/.js)...
[info] scalafmt: Reformatted 1 Scala sources
[success] Total time: 2 s, completed Jan 12, 2022, 12:53:46 PM
imkarrer@IsaacsWorkstation woof % 
```

Comment (hejfelix) [reply modules/slf4j/src/main/scala/org/legogroup/woof/slf4j/WoofLogger.scala:19]: That's...interesting, let me have a look 😄

**Comment by hejfelix** [comment]
Thanks, I fixed it by setting trailing commas to "keep"

**Pull request merged**